src/toil/wdl/wdltoil.py

```diff
@@ -911,14 +995,24 @@ def devirtualize_to(filename: str, dest_dir: str, file_source: Union[AbstractFil
                 virtualized_to_devirtualized[filename] = result
             logger.debug('Devirtualized %s as openable file %s', filename, result)
         else:
-            # This is a local file
-            # To support relative paths, join the execution dir and filename
-            # if filename is already an abs path, join() will do nothing
-            if execution_dir is not None:
-                result = os.path.join(execution_dir, filename)
-            else:
-                result = filename
-            logger.debug("Virtualized file %s is already a local path", filename)
+            result = None
+            if wdl_options.get("filename_to_uri") is not None:
+                # We support importing files relative to a URL when the input JSON itself is downloaded
+                # See import_url_files()
+                filename_uri = wdl_options["filename_to_uri"].get(filename)
```

Comment:
I think there's a missing piece of the explanation here. *Because* we can import workflow inputs by relative URL, we end up with those relative URLs stored somewhere (where they can become `filename` here I guess?), so a filename can *look* like it is a local file but *actually* be meant to be a URL, so we have to go get the URL for it to actually use?

But then what if `inputs.json` has `"WorkflowName.the_file": "file.txt"`, and comes from a URL where we want to fetch the adjacent "file.txt", but then the WDL itself has `File the_other_file = "file.txt"` at workflow scope, which we would expect to read from the current working directory. Won't we not be able to keep them straight?

And since we pass the one `wdl_options` through the whole workflow, won't this case get hit inside a task output section when we want an output that happens to be named "file.txt"?

Reply:
Fixed with a new implementation